### ISSUE TITLE
Query: Make ParameterExtraction recursive on closure members

### ISSUE THREAD

**Issue by smitpatel** [aspnet/EntityFrameworkCore#9105]
For queries like
```C#
var i = 1;
Expression<Func<Child, bool>> whereExpression = c => c.Id == i;
db.Children.Where(whereExpression).ToList();
```
`whereExpression` & `i` inside of it both will appear as closure members in the ExpressionTree.
In our ParameterExtractingExpressionVisitor, we unwrap `whereExpression` from the closure but we don't process it further so we leave closure.i in the ExpressionTree, which is then converted to ConstantExpression by ReLinq query parser hence we miss opportunity to parametrize.

This issue is to port fix for #8909 to patch.

**Comment by ajcvickers** [comment]
@smitpatel What are the consequences of not parameterizing? Is it only cache misses in the query cache?

**Comment by smitpatel** [comment]
@ajcvickers - Yes.

**Comment by Eilon** [comment]
This patch is *approved* for 1.1.x. Please follow the patch creation instructions that were sent by email, and submit a PR to the correct branch.

@ajcvickers - please assign.

**Comment by smitpatel** [comment]
Assigning to @maumar To do porting and introduce a quirk. My machine does not have VS 2015 and cannot build 1.1.3 patch branch.

**Comment by divega** [comment]
Fixed in c19c8c25f9b9e56a91687d1320edbd042726c4e7.
